src/java.base/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -339,6 +340,36 @@ static void checkType(Class<?> actualType, Class<?> expectedType) {
         }
     }
 
+    public static boolean isPowerOfTwo(int width) {
+        return Integer.bitCount(width) == 1;
+    }
+
+    public static int byteWidthOfPrimitive(Class<?> type) {
```

Comment:
When we create memory access var handles, we use this to compute the byte size of a primitive (`VarHandles::memorySegmentViewHandle`):

```
Wrapper.forPrimitiveType(carrier).bitWidth() / 8
```

Perhaps we should put this into Utils and then reuse in both places?

Reply:
Good idea. Done.